pkg/os/volume/api.go

```diff
@@ -54,56 +57,67 @@ var (
 	// PS C:\disks> (Get-Disk -Number 1 | Get-Partition | Get-Volume).UniqueId
 	// \\?\Volume{452e318a-5cde-421e-9831-b9853c521012}\
 	VolumeRegexp = regexp.MustCompile(`Volume\{[\w-]*\}`)
+	// longPathPrefix is the prefix of Windows long path
+	longPathPrefix = "\\\\?\\"
+
+	notMountedFolder = errorspkg.New("not a mounted folder")
 )
 
 // New - Construct a new Volume API Implementation.
 func New() VolumeAPI {
 	return VolumeAPI{}
 }
 
-func getVolumeSize(volumeID string) (int64, error) {
-	cmd := `(Get-Volume -UniqueId "$Env:volumeID" | Get-partition).Size`
-	cmdEnv := fmt.Sprintf("volumeID=%s", volumeID)
-	out, err := utils.RunPowershellCmd(cmd, cmdEnv)
-
-	if err != nil || len(out) == 0 {
-		return -1, fmt.Errorf("error getting size of the partition from mount. cmd %s, output: %s, error: %v", cmd, string(out), err)
+// ListVolumesOnDisk - returns back list of volumes(volumeIDs) in a disk and a partition.
+func (VolumeAPI) ListVolumesOnDisk(diskNumber uint32, partitionNumber uint32) (volumeIDs []string, err error) {
+	partitions, err := cim.ListPartitionsOnDisk(diskNumber, partitionNumber, []string{"ObjectId"})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list partition on disk %d", diskNumber)
```

Comment:
I believe this would be on top of "errors" and not "github.com/microsoft/wmi/pkg/errors" after the refactor.